networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManager.java

```diff
@@ -91,17 +94,40 @@ protected SafeFuture<?> doStart() {
     synchronized (this) {
       staticPeers.forEach(this::createPersistentConnection);
     }
-    periodicPeerSearch =
-        asyncRunner.runWithFixedDelay(
-            this::searchForPeers,
-            DISCOVERY_INTERVAL,
-            error -> LOG.error("Error while searching for peers", error));
-    connectToBestPeers(emptyList());
-    searchForPeers();
+    startSearchPeerTask();
     peerConnectedSubscriptionId = network.subscribeConnect(this::onPeerConnected);
     return SafeFuture.COMPLETE;
   }
 
+  @SuppressWarnings("FutureReturnValueIgnored")
```

Comment:
We shouldn't be ignoring these returned futures - it would silently ignore any exceptions they complete with.